src/main/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessor.java

```diff
@@ -48,4 +49,13 @@ public void doExecute(
             handler.accept(ingestDocument, null);
         }, e -> { handler.accept(null, e); }));
     }
+
+    @Override
+    public void doBatchExecute(List<String> inferenceList, Consumer<List<?>> handler, Consumer<Exception> onException) {
+        mlCommonsClientAccessor.inferenceSentences(
+            this.modelId,
+            inferenceList,
+            ActionListener.wrap(vectors -> { handler.accept(vectors); }, exception -> onException.accept(exception))
```

Comment:
Just use `onException` here instead of `exception -> onException.accept(exception)`